presto-main/src/main/java/io/prestosql/sql/planner/optimizations/HashGenerationOptimizer.java

```diff
@@ -620,7 +620,14 @@ public PlanWithProperties visitUnion(UnionNode node, HashComputationSet parentPr
         public PlanWithProperties visitProject(ProjectNode node, HashComputationSet parentPreference)
         {
             Map<Symbol, Symbol> outputToInputMapping = computeIdentityTranslations(node.getAssignments().getMap());
-            HashComputationSet sourceContext = parentPreference.translate(symbol -> Optional.ofNullable(outputToInputMapping.get(symbol)));
+            ImmutableMap.Builder<HashComputation, HashComputation> builder = ImmutableMap.builder();
+            for (HashComputation hashComputation : parentPreference.getHashes()) {
+                hashComputation.translate(symbol -> Optional.ofNullable(outputToInputMapping.get(symbol)))
+                        .ifPresent(x -> builder.put(x, hashComputation));
+            }
+            Map<HashComputation, HashComputation> hashComputationMap = builder.build();
```

Comment:
This needs a better name. First, the `Map` part in the name is redundant, since it's clear it's a map from the type declaration. But more importantly, what does this mapping represent? What do the keys and values represent?

Reply:
also it could be inlined, then `builder` could get that "better" name